internal/sync/syncer_related.go

```diff
@@ -35,13 +35,15 @@ import (
 
 func (s *ResourceSyncer) processRelatedResources(log *zap.SugaredLogger, stateStore ObjectStateStore, remote, local syncSide) (requeue bool, err error) {
 	for _, relatedResource := range s.pubRes.Spec.Related {
-		requeue, err := s.processRelatedResource(log.With("identifier", relatedResource.Identifier), stateStore, remote, local, relatedResource)
-		if err != nil {
-			return false, fmt.Errorf("failed to process related resource %s: %w", relatedResource.Identifier, err)
-		}
+		if !relatedResource.Optional {
```

Comment:
Doesn't this mean that optional resources are not synced _at all_ anymore?

Reply:
Correct, but if that was not the point of the issue and as you mentioned we have this piece
https://github.com/kcp-dev/api-syncagent/blob/c48b2155cb0fe3d5e656723ff44f20d0e8bf52b1/internal/sync/syncer_related.go#L84-L92
Let's maybe drop the Optional field and add the tests instead

Reply:
I think the original issue, after Marvin explained my own code to me, was that the evaluating of the references yielded an error if the path in the object doesn't exist (yet), like if `spec.secret.name.whatever` just doesn't exist yet. _That_ is what we should simply handle and ignore.